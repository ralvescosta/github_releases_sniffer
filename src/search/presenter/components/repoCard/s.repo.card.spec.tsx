import React from 'react';
import {render} from '@testing-library/react-native';

import {RepoCard} from './index';
import {RepoCardViewControllerSpy} from '../../__test__/repo.card.view.controller.spy';

function makeSut() {
  const Wrapper = () => {
    const repoController = new RepoCardViewControllerSpy();
    return <RepoCard viewController={repoController} />;
  };

  return render(<Wrapper />);
}

describe('Search Repo Card Components', () => {
  it('Should render correctly', () => {
    const sut = makeSut();

    expect(sut.toJSON()).toMatchSnapshot();
  });
});
