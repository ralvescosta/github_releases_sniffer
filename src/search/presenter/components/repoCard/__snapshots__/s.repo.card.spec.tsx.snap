// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Repo Card Components Should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 6,
      "elevation": 5,
      "flexDirection": "row",
      "marginTop": 15,
      "padding": 15,
      "width": "95%",
    }
  }
>
  <View
    style={
      Object {
        "maxWidth": "60%",
        "width": "100%",
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text
        numberOfLines={2}
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Object {
            "color": "#777",
            "marginTop": 4,
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "maxWidth": "40%",
        "width": "100%",
      }
    }
  >
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      value={false}
    />
  </View>
</View>
`;
